routing/router.go

```diff
@@ -1609,7 +1613,23 @@ func (r *ChannelRouter) SendPayment(payment *LightningPayment) ([32]byte, *route
 		return [32]byte{}, nil, err
 	}
 
-	return r.sendPayment(payment, paySession)
+	// Record this payment hash with the ControlTower, ensuring it is not
+	// already in-flight.
+	info := &channeldb.CreationInfo{
+		PaymentHash:    payment.PaymentHash,
+		Value:          payment.Amount,
+		CreationDate:   time.Now(),
+		PaymentRequest: nil,
+	}
+
+	err = r.cfg.Control.InitPayment(payment.PaymentHash, info)
+	if err != nil {
+		return [32]byte{}, nil, err
+	}
+
+	// Since this is the first time this payment is being made, we pass nil
+	// for the existing attempt.
+	return r.sendPayment(nil, payment, paySession)
```

Comment:
Why can't we do this in the same DB transaction as the initial creation?

Reply:
Not sure what you mean? `existingAttempt==nil` indicates to the payment loop that it should attempt a new attempt immediately.